torch/__init__.py

```diff
@@ -270,22 +270,30 @@ def set_default_tensor_type(t):
 
 
 def set_default_dtype(d):
-    r"""Sets the default floating point dtype to :attr:`d`. This type will be
-    used as default floating point type for type inference in
-    :func:`torch.tensor`.
+    r"""Sets the default floating point dtype to :attr:`d`.
+    This dtype is:
+    1. The inferred dtype for python floats in :func:`torch.tensor`.
```

Comment:
This isn't quite right since it's the inferred type for Python floats in other contexts, too:

```
torch.set_default_dtype(torch.double)
(t + 5.).dtype
: torch.float64
```

Reply:
I think this falls under the `torch.tensor` case since when you add a scalar to a tensor, we first make a zero dim tensor containing the scalar (hence the dtype of this new tensor in this case will be `torch.float64`

Reply:
That's true in the first case but also an implementation detail I wouldn't expect a user to be be aware of. In the second case I don't think `torch.tensor()` is involved?